frontend/src/app/user/service/user-role.service.ts

```diff
@@ -0,0 +1,24 @@
+import {Injectable} from '@angular/core';
+import {ActivatedRouteSnapshot, RouterStateSnapshot} from '@angular/router';
+import {UserByEmailResolverService} from './user-by-email-resolver.service';
+import {Observable} from 'rxjs/Observable';
+
+@Injectable()
+export class UserRoleService {
+
+    constructor(private userByEmailResolverService: UserByEmailResolverService) { }
+
+    /**
+     * determine if the current user has any role in list
+     */
+    hasRoles(list: [string]): Observable<boolean> {
+        return this.userByEmailResolverService.resolve().flatMap((v) => {
+            const roles = v._embedded.users[0].authorities.map(r => r.authority);
```

Comment:
What is `r`? Is `r` a `role`? It's unclear from the code, but would probably be cleared up looking at API documentation.